Assets/LeapMotion/Core/Scripts/Hands/RiggedFinger.cs

```diff
@@ -12,40 +12,87 @@
 using Leap;
 
 namespace Leap.Unity {
-  /**
-* Manages the orientation of the bones in a model rigged for skeletal animation.
-* 
-* The class expects that the graphics model bones corresponding to bones in the Leap Motion 
-* hand model are in the same order in the bones array.
-*/
-  public class RiggedFinger : FingerModel {
 
-    /** Allows the mesh to be stretched to align with finger joint positions
-     * Only set to true when mesh is not visible
-     */
+  /// <summary>
+  /// Manages the position and orientation of the bones in a model rigged for skeletal
+  /// animation.
+  ///  
+  /// The class expects that the graphics model's bones that correspond to the bones in
+  /// the Leap Motion hand model are in the same order in the bones array.
+  /// </summary>
+  public class RiggedFinger : FingerModel {
+    
+    /// <summary>
+    /// Allows the mesh to be stretched to align with finger joint positions.
+    /// Only set to true when mesh is not visible.
+    /// </summary>
     [HideInInspector]
     public bool deformPosition = false;
 
+    [HideInInspector]
+    public bool scaleLastFingerBone = false;
+
     public Vector3 modelFingerPointing = Vector3.forward;
     public Vector3 modelPalmFacing = -Vector3.up;
 
     public Quaternion Reorientation() {
       return Quaternion.Inverse(Quaternion.LookRotation(modelFingerPointing, -modelPalmFacing));
     }
+    
+    /// <summary> Backing store for s_standardFingertipLengths, don't touch. </summary>
+    private static float[] s_backingStandardFingertipLengths = null;
+    /// <summary>
+    /// Lazily-calculated fingertip lengths for the standard edit-time hand.
+    /// </summary>
+    private static float[] s_standardFingertipLengths {
+      get {
+        if (s_backingStandardFingertipLengths == null) {
```

Comment:
What value does that add? What if the consuming application never uses a RiggedFinger? In general, I prefer lazy initialization because you only ever pay for what you use. But I think it's an interesting discussion too.

Reply:
Value would be less variables, no need for a property, simpler in general.  Also the way C# works is that a class is not initialized unless it is used!  So in a way, you can use a static constructor as a lazy initializer for all static data of the class!

Reply:
I still prefer lazy initialization because as a general strategy it's more robust to, for example, order-of-execution issues. We've also been burned in the past by Unity and C# "guarantees" :P 
But I moved this logic to a static initializer.